src/net.cpp

```diff
@@ -895,20 +895,7 @@ size_t CConnman::SocketSendData(CNode& node) const
  */
 bool CConnman::AttemptToEvictConnection()
 {
-    std::vector<NodeEvictionCandidate> vEvictionCandidates;
-    {
-        LOCK(m_nodes_mutex);
-        for (const CNode* node : m_nodes) {
-            if (node->fDisconnect)
```

Comment:
This removes the check `node->fDisconnect`, which I think could be dangerous:
We set `fDisconnect` in `AttemptToEvictConnection()`, but don't immediately disconnect. The actual cleanup happens at some later point, when `ThreadSocketHandler` calls `DisconnectNodes()`.
So if we have an attacker that makes new connections to us rapidly, there would be a race between `ThreadSocketHandler` and the attacker making the next connection to us. If the attacker wins this race, we could accept two inbound connections, but, without the check that is being removed here, `AttemptToEvictConnection()` could mark the same inbound peer for eviction twice, so that we'll actually only evict one peer. Since inbound eviction is triggered only in `CreateNodeFromAcceptedSocket()` (and not through some regularly scheduled job), this could lead to us having an extra inbound peer above our limit indefinitely. 
Now if the attacker repeats this scheme many times, they could make thousands of inbound connections to us / exhaust our resources / crash our node.

Reply:
hmm, thinking more about this, I didn't see the whole picture, because everything happens in `ThreadSocketHandler` - after ` CConnman::SocketHandler() -> CConnman::SocketHandlerListening()`, `CConnman::DisconnectNodes()` is called, so it's not possible to accept new connections in between - however,  `CConnman::SocketHandlerListening()` calls `AcceptConnection()` in a loop for multiple `ListenSocket`s, so if we have multiple listening sockets and an attacker would connect to more than one at the same time, the scenario above may still be possible?